common/src/main/java/tds/common/cache/CacheKeyGenerator.java

```diff
@@ -0,0 +1,34 @@
+package tds.common.cache;
+
+import org.springframework.cache.interceptor.KeyGenerator;
+
+import java.lang.reflect.Method;
+
+/**
+ * A custom key generator for caching
+ */
+public class CacheKeyGenerator implements KeyGenerator {
+    @Override
+    public Object generate(Object target, Method method, Object... params) {
+        return this.generate(target.getClass().getName(), method.getName(), params);
+    }
+
+    public String generate(String className, String methodName, Object... params) {
+        StringBuilder sb = new StringBuilder();
```

Comment:
You can chain these so it'd be `StringBuilder sb = new StringBuilder(className).append(".").append(methodName);`